src/attribute.rs

```diff
@@ -409,6 +411,34 @@ impl<'a> Attribute<'a> {
                     &buf[..pos],
                 ))))
             }
+            NC_STRING => {
+                let mut buf: Vec<*mut c_char> = vec![std::ptr::null_mut(); attlen];
+                let result;
+                unsafe {
+                    error::checked(super::with_lock(|| {
+                        nc_get_att_string(
+                            self.ncid,
+                            self.varid,
+                            self.name.as_ptr() as *const _,
+                            buf.as_mut_ptr() as *mut *mut _,
+                        )
+                    }))?;
+
+                    result = buf
+                        .iter()
+                        .map(|cstr_pointer| {
+                            if cstr_pointer.is_null() {
+                                String::new()
+                            } else {
+                                CStr::from_ptr(*cstr_pointer).to_string_lossy().to_string()
+                            }
+                        })
+                        .collect();
+
+                    nc_free_string(attlen, buf.as_mut_ptr());
```

Comment:
You must `super::with_lock` this

Reply:
Does it make sense to hold the lock while copying the data over to rust or should we take it again here?

Reply:
No, it is only needed when calling functions from `netcdf-c`, we should have sole ownership of the data following `nc_get_att_string`

Reply:
We can also ignore the error handling in this case: We don't care about a potential memory leak

Reply:
Looks like performing several fallible operations in one `with_lock` invocation is difficult, thinking about adding another helper

Reply:
You only have to wrap `nc_free_string`, everything else is fine here. The lock is just there to synchronise `nc_*` calls

Reply:
Does this make sense?
https://github.com/georust/netcdf/pull/85/commits/bc7b89d0ccd73589b75b2ecf43242485f244cd99

Reply:
Ah ok, changing to that

Reply:
Looks like it always returns 0
https://www.unidata.ucar.edu/software/netcdf/docs/dvar_8c_source.html#l01206

Reply:
No worries with this then :)